src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int SIZE_ARRAY = 10;
+    private static final double COEFFICIENT_SIZE_ARRAY = 1.5;
+    private T[] value;
+    private int size;
+
+    public ArrayList() {
+        this.value = (T[]) new Object[SIZE_ARRAY];
+        this.size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        widenArray();
+        this.value[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        widenArray();
+        System.arraycopy(this.value, index, this.value, index + 1, size - index);
+        this.value[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index > size - 1 || index < 0) { // if we use a method checkIndex => +1 error;
+            throw new ArrayIndexOutOfBoundsException("Houston we have a BUG!");
+        }
+        return value[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index > size - 1 || index < 0) { // if we use a method checkIndex => +1 error;
```

Comment:
```suggestion
        if (index >= size || index < 0) { // if we use a method checkIndex => +1 error;
```

Reply:
remove comments. you can move this check to boolean method and use in set(), get() remove(). In add method, you can use the same check as in checkIndex() method.

Reply:
done